src/test/groovy/org/prebid/server/functional/tests/StoredResponseSpec.groovy

```diff
@@ -249,4 +252,128 @@ class StoredResponseSpec extends BaseSpec {
+        bidRequest.imp[0].ext.prebid.storedAuctionResponse = new StoredAuctionResponse().tap {
+            id = PBSUtils.randomString
+            seatBidObject = storedAuctionResponse
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should contain same stored auction response as requested"
+        assert convertToComparableSeatBid(response.seatbid) == [storedAuctionResponse]
+
+        and: "PBS not send request to bidder"
+        assert bidder.getRequestCount(bidRequest.id) == 0
+    }
+
+    def "PBS should set seatBids in response from multiple imp.ext.prebid.storedBidResponse.seatbidobj when it is defined"() {
+        given: "BidRequest with multiple imps"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp = [impWithSeatBidObject, impWithSeatBidObject]
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should contain same stored auction response bids as requested"
+        assert convertToComparableSeatBid(response.seatbid).bid.flatten().sort() ==
+                bidRequest.imp.ext.prebid.storedAuctionResponse.seatBidObject.bid.flatten().sort()
+
+        and: "PBS not send request to bidder"
+        assert bidder.getRequestCount(bidRequest.id) == 0
+    }
+
+    def "PBS should prefer seatbidarr from request over seatbidobj from imp when both are present"() {
+        given: "Default basic BidRequest with stored response"
+        def bidRequest = BidRequest.defaultBidRequest
+        def storedAuctionResponse = SeatBid.getStoredResponse(bidRequest)
+        bidRequest.tap{
+            imp[0].ext.prebid.storedAuctionResponse = new StoredAuctionResponse().tap {
+                seatBidObject = SeatBid.getStoredResponse(bidRequest)
+            }
+            ext.prebid.storedAuctionResponse = new StoredAuctionResponse(seatBids: [storedAuctionResponse])
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should contain same stored auction response as requested"
+        assert response.seatbid == [storedAuctionResponse]
+
+        and: "PBS not send request to bidder"
+        assert bidder.getRequestCount(bidRequest.id) == 0
+    }
+
+    private static final Imp getImpWithSeatBidObject() {
+        def imp = Imp.defaultImpression
+        def bids = Bid.getDefaultBids([imp])
+        def seatBid = new SeatBid(bid: bids, seat: GENERIC)
+        imp.tap {
+            ext.prebid.storedAuctionResponse = new StoredAuctionResponse(seatBidObject: seatBid)
+        }
+    }
+
+    private static final List<SeatBid> convertToComparableSeatBid(List<SeatBid> seatBid) {
+        seatBid*.tap {
+            it.bid*.ext = null
+            it.group = null
+        }
+        seatBid
```

Comment:
seatBid redundant.

```
 seatBid*.tap {
            it.bid*.ext = null
            it.group = null
        }
```
seatBid with tap should be enough

Reply:
good catch, updated